app/javascript/spec/namespace-form/namespace-form.spec.js

```diff
@@ -0,0 +1,148 @@
+  const flashSpy = jest.spyOn(window, 'add_flash');
+
+  const defaultProps = {
+    isDomain: true,
+    namespaceId: 2222,
+    parentId: null,
+  };
+
+  afterEach(() => {
+    fetchMock.restore();
+  });
+
+  it('should correctly create the schema for domain', () => {
+    const expectedResult = [
+      expect.objectContaining({
+        component: 'text-field',
+        isReadOnly: false,
+        label: 'Name:',
+        name: 'name',
+      }),
+      expect.objectContaining({
+        component: 'text-field',
+        isReadOnly: false,
+        label: 'Description:',
+        name: 'description',
+      }),
+      expect.objectContaining({
+        component: 'switch-field',
+        label: 'Enabled:',
+        name: 'enabled',
+      })];
+
+    const { fields } = schema(true, false);
+    expect(fields).toEqual(expectedResult);
+  });
+
+  it('should correctly initialize the component for namespace', (done) => {
+    const initialProps = {
+      isDomain: false,
+      namespaceId: 1111,
+      parentId: 2222,
+    };
+
+    fetchMock
+      .getOnce('/api/automate_namespaces/1111?attributes=name,description,enabled,source',
+        {
+          name: 'f', description: 'a', enabled: null, source: 'user',
+        });
+
+    const wrapper = mount(<NamespaceForm {...initialProps} />);
+    setImmediate(() => {
+      wrapper.update();
+      expect(wrapper.find('label').first().text()).toEqual('Name:');
+      expect(wrapper.find('input').first().instance().value).toEqual('f');
+      done();
+    });
+  });
+
+  it('should correctly initialize the component for domain', (done) => {
+    fetchMock
+      .getOnce('/api/automate_domains/2222?attributes=name,description,enabled,source',
+        {
+          name: 's', description: 'z', enabled: true, source: 'user',
+        });
+
+    const wrapper = mount(<NamespaceForm {...defaultProps} />);
+    setImmediate(() => {
+      wrapper.update();
+      expect(wrapper.find('input').first().props().value).toEqual('s');
+      expect(wrapper.find('input').last().props().value).toBeTruthy();
+      done();
+    });
+  });
+
+  it('should send correct data on submit', (done) => {
+    const initialProps = {
+      isDomain: false,
+      namespaceId: null,
+      parentId: 2222,
+    };
+    fetchMock.getOnce('/api/automate_domains/2222?attributes=name,description,enabled,source',
+      {
+        name: 's', description: 'z', enabled: true, source: 'user',
+      }).post('/api/automate_namespaces/', {});
+
+    const wrapper = mount(<NamespaceForm {...initialProps} />);
+    setImmediate(() => {
+      const Form = wrapper.find(FormRenderer).childAt(0);
+      Form.instance().form.change('name', 'New');
+      Form.instance().form.change('description', 'This is really New');
+      wrapper.update();
+      wrapper.find('button').at(0).simulate('click');
+      setImmediate(() => {
+        // I have no idea how to test it. The Object.defineProperty is not working.
```

Comment:
You can mock the `POST` API as well and they you can check what values was send to it

Please, look on fetchMock docs or other form tests

Reply:
Which part of the fetchMock please?

Reply:
http://www.wheresrhys.co.uk/fetch-mock/#api-mockingget_post

Reply:
You can check the `fetchMock.lastCall()`. It will return all you need to know about the request.
wheresrhys.co.uk/fetch-mock/#api-inspectionlastCall